File: backend/app/schemas/contract.py
from pydantic import BaseModel, Field
from datetime import date, time
from typing import Optional
from enum import Enum

class RecurrenceType(str, Enum):
    WEEKLY = "weekly"
    BIWEEKLY = "biweekly"
    MONTHLY = "monthly"

class ContractCreate(BaseModel):
    venue_id: int
    start_date: date
    end_date: date
    day_of_week: int = Field(..., ge=0, le=6)
    start_time: time
    recurrence: RecurrenceType = RecurrenceType.WEEKLY
    discounted_price: int = Field(..., gt=0)
    description: Optional[str] = None

class ContractResponse(BaseModel):
    id: int
    venue_id: int
    start_date: date
    end_date: date
    day_of_week: int
    start_time: time
    recurrence: RecurrenceType
    original_price: int
    discounted_price: int
    total_amount: int
    status: str
    description: Optional[str]
    
    class Config:
        from_attributes = True

class ContractSessionCancel(BaseModel):
    contract_id: int
    session_date: date
    reason: str

File: frontend/src/types/contract.ts
export type ContractStatus = 'active' | 'expired' | 'cancelled'
export type RecurrenceType = 'weekly' | 'biweekly' | 'monthly'

export interface Contract {
  id: number
  venueId: number
  venueName: string
  startDate: string
  endDate: string
  recurrence: RecurrenceType
  dayOfWeek: number
  startTime: string
  duration: number
  sessionsCount: number
  pricePerSession: number
  totalAmount: number
  status: ContractStatus
  description?: string
  createdAt: string
}

export interface ContractCreate {
  venueId: number
  startDate: string
  endDate: string
  recurrence: RecurrenceType
  dayOfWeek: number
  startTime: string
  pricePerSession: number
  description?: string
}

export interface ContractUpdate extends Partial<ContractCreate> {
  status?: ContractStatus
}

export interface ContractSession {
  date: string
  isAttended: boolean
  isCancelled: boolean
  cancellationReason?: string
}